pkg/services/ngalert/state/persister_sync.go

```diff
@@ -107,7 +107,7 @@ func (a *SyncStatePersister) saveAlertStates(ctx context.Context, states ...Stat
 			ResultFingerprint: s.ResultFingerprint.String(),
 		}
 
-		err = a.store.SaveAlertInstance(ctx, instance)
+		err = a.store.SaveAlertInstances(ctx, []models.AlertInstance{instance})
```

Comment:
The `SyncStatePersister` supports `maxStateSaveConcurrency` and the code around it I didn't want to change in this refactoring, so here it just calls `SaveAlertInstances` with one alert instance at a time, keeping the old behaviour.